pytorch_toolkit/formula_recognition/tools/utils/exporter.py

```diff
@@ -0,0 +1,127 @@
+
+    def export_decoder(self):
+        tgt = np.array([[START_TOKEN]] * 1)
+        decoder_inputs = self.config.get("decoder_input_names", DECODER_INPUTS).split(',')
+        decoder_outputs = self.config.get("decoder_output_names", DECODER_OUTPUTS).split(',')
+        row_enc_out = torch.rand(FEATURES_SHAPE)
+        hidden = torch.randn(HIDDEN_SHAPE)
+        context = torch.rand(CONTEXT_SHAPE)
+        output = torch.rand(OUTPUT_SHAPE)
+
+        torch.onnx.export(self.decoder,
+                          (hidden,
+                           context,
+                           output,
+                           row_enc_out,
+                           torch.tensor(tgt, dtype=torch.long)),
+                          self.config.get("res_decoder_name"),
+                          opset_version=11,
+                          input_names=decoder_inputs,
+                          output_names=decoder_outputs,
+                          dynamic_axes={decoder_inputs[3]: {  # row_enc_out name should be here
+                              0: 'batch', 1: 'H', 2: 'W'}}
+                          )
+
+    def export_encoder_ir(self):
+        export_command = f"""{OPENVINO_DIR}/bin/setupvars.sh && \
+        python {OPENVINO_DIR}/deployment_tools/model_optimizer/mo.py \
+        --framework onnx \
+        --input_model {self.config.get("res_encoder_name")} \
+        --input_shape "{self.config.get("input_shape_decoder")}" \
+        --output "{self.config.get("encoder_output_names", ENCODER_OUTPUTS)}" \
+        --reverse_input_channels \
+        --scale_values 'imgs[255,255,255]'"""
+        if self.config.get('verbose_export'):
+            print(export_command)
+        subprocess.run(export_command,
+                       shell=True, check=True
+                       )
+
+    def export_decoder_ir(self):
+        input_shape_decoder = self.config.get("input_shape_decoder")
+        output_h, output_w = input_shape_decoder[2] / 32, input_shape_decoder[3] / 32
+        if self.config['backbone_config']['disable_layer_4']:
+            output_h, output_w = output_h * 2, output_w * 2
+        if self.config['backbone_config']['disable_layer_3']:
+            output_h, output_w = output_h * 2, output_w * 2
+        output_h, output_w = math.ceil(output_h), math.ceil(output_w)
+        input_shape = [[1, self.config.get('head', {}).get("decoder_hidden_size", 512)],
+                       [1, self.config.get('head', {}).get('decoder_hidden_size', 512)],
+                       [1, self.config.get('head', {}).get('encoder_hidden_size', 256)],
+                       [1, output_h, output_w, self.config.get('head', {}).get('decoder_hidden_size', 512)], [1, 1]]
+        input_shape = "{}, {}, {}, {}, {}".format(*input_shape)
+        export_command = f"""{OPENVINO_DIR}/bin/setupvars.sh &&
+        python {OPENVINO_DIR}/deployment_tools/model_optimizer/mo.py \
+        --framework onnx \
+        --input_model {self.config.get('res_decoder_name')} \
```

Comment:
Please, if it is possible, move codes like `self.config.get("key", default_val)` outside the f-strings to separate variables.

It is not mandatory, but IMHO it will really improve code style -- at least it will make it a bit more readable and many editors will make proper highlighting for it.

(NB: adding proper variables with short understandable names often can significantly improve code readability)

Reply:
Refactored